### PR TITLE
Initial Setup with ESlint, Prettier, TailwindCSS

### LINKED ISSUE #1: Base Setup
### Requirements
- Setup the repo with:
    - Next.JS (With Pages Dir)
    - Storybook
    - React Testing Library
    -  Tanstack query
    - Eslint
    - Prettier
    - Tailwind CSS
    - React hook form
    - Zod
    - Zustand
- Add a static export build script to allow us to host the app on Cloudflare pages.
- Setup Github actions:
   - To host the changes on Cloudflare pages when changes are merged to the develop branch.
   - To host the changes on Cloudflare pages when changes are merged to the main branch. 

### Context for adding static export to the app
- As we will not be using any Nest.JS API routes (as we have our own backend) the webapp can be hosted as a static build on Cloudflare pages.

### PR DESCRIPTION
**Pull request by yesyash**
### Changes made
- Set up the project with Estlint and import order to have a consistent order import. 
- Add tailwind for styling & prettier for formatting
- Add Code editor settings to allow for a consistent Dev experience.
- Fix `end of the line` to always use `lf` and not `crlf`.

### CODE REVIEW

**Review by satyam73 — DISMISSED** [review]
LGTM

**Review by satyam73 — APPROVED** [review]

**Review by shubhamsinghbundela — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "prettier-fix": "prettier --write ."
+  },
+  "dependencies": {
+    "class-variance-authority": "^0.7.0",
+    "next": "13.4.19",
+    "postcss": "8.4.28",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "tailwind-merge": "^1.14.0",
+    "typescript": "5.1.6"
```
Comment (iRohitGaur) [inline package.json:20]: `postcss`, `tailwind-merge`, and `typescript` should go in devDependencies. Check once for cva as well.

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { AppProps } from "next/app";
+import { Inter } from "next/font/google";
+
+import { cn } from "@/utils/classname";
+import "@/styles/globals.css";
+
+const inter = Inter({ subsets: ["latin"], weight: ["400", "500", "600", "700"], variable: "--font-inter" });
```
Comment (iRohitGaur) [inline src/pages/_app.tsx:7]: suggestion: this can be handled directly in `globals.css`

File: src/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import Homepage from "../modules/homepage";
```
Comment (iRohitGaur) [inline src/pages/index.tsx:1]: use `@` everywhere for consistency

**Review by iamitprakash — APPROVED** [review]

File: src/pages/index.tsx (hunk 3 above)
Comment (yesyash) [reply src/pages/index.tsx:1]: - As it is just one folder back @ feels like overkill to me as you are going back to the src folder and then importing the file which was just a folder back.

File: src/pages/_app.tsx (hunk 2 above)
Comment (yesyash) [reply src/pages/_app.tsx:7]: - We're using `next/font` here to import the `inter` font and this is the way they suggest it to be used in the [docs](https://nextjs.org/docs/pages/building-your-application/optimizing/fonts). 
We can handle this in `global.css` if we were not using `next/font` and stored all the font files in public dir. 

Did I miss something?

File: package.json (hunk 1 above)
Comment (yesyash) [reply package.json:20]: - Made a PR for this: #5 
- CVA is a package that we need at runtime so not moving to dev dependencies.

File: src/pages/_app.tsx (hunk 2 above)
Comment (iRohitGaur) [reply src/pages/_app.tsx:7]: No, it's not a wrong approach. It's just that I personally prefer to keep this part in globals file.

**Review by iRohitGaur — APPROVED** [review]

**Review by vvaibhavdesai — APPROVED** [review]

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{
+  /**
+    Editor rules
+    ------------
+  */
```
Comment (prakashchoudhary07) [inline .vscode/settings.json:5]: Does `.json` support comments?

**Pull request merged**